gaeguli/types.h

```diff
@@ -48,6 +48,7 @@ typedef enum {
   GAEGULI_VIDEO_SOURCE_V4L2SRC,
   GAEGULI_VIDEO_SOURCE_AVFVIDEOSRC,
   GAEGULI_VIDEO_SOURCE_VIDEOTESTSRC,
+  GAEGULI_VIDEO_SOURCE_MANUAL = G_MAXINT,
```

Comment:
Do we really need to add the `_MANUAL` items into `GaeguliVideoSource` and `GaeguliEncodingMehod`? Wouldn't it be easier to just check if `self->source_pipeline` and/or `self->encoding_pipeline` are `NULL` when instantiating the pipeline in `_build_target_pipeline()`?

Reply:
Yes, it would be. 
Actually, I was thinking about exposing internal pipeline string via properties for the other cases. But I feel it makes complex tot handle properties; e.g only  `MANUAL` allows to set the properties. for other enum values, `source_pipeline` and `encoding_pipeline` shouldn't be set even though they are writable. 

Perhaps, I should look for nicer way to support various pipelines (various devices)